example/src/Examples/API/Images.tsx

```diff
@@ -38,6 +43,22 @@ export const Images = () => {
   const oslo = useImage(require("../../assets/oslo.jpg"));
   const coffee = useImage("https://picsum.photos/id/1060/640/360");
 
+  // Verify that "useImage" works even when source change
+  const [asyncUrl, setAsyncUrl] = useState<string>();
+
+  // Verify that "useImage" works even when the source change after initial rendering
+  // It could be because the source is fetched remotely, or defined only after some interactions...
+  const asyncImage = useImage(asyncUrl);
```

Comment:
I don't think that we want to support this at all. @chrfalch do you agree?

Reply:
Not support as an example, you mean? 🤔
 
This is purely the use-case described in #571, where the source I pass to `useImage()` is undefined at first render, but becomes defined once the user selects an image to use among several

Reply:
We want to support having undefined as a parameter, but we don't want to support the source changing over time.

Reply:
I think it should be supported - it isn't uncommon to change the source of an image in React Native so developers will probably use the same paradigm here as well.

Reply:
I would love to see some examples/use-cases

Reply:
> I would love to see some examples/use-cases

I'm using skia in an app where a user can modify an image from the gallery. In this app, the loaded image will have multiple versions. This is because the image will be worked on in Skia and sent to a server for processing and then sent back to the user. The processed image will need to be displayed, this is not possible with useImage because it latches onto the initial image. Thus, the user cannot keep working on this image. Another example is that when the user wants to change image by loading another one. For the new image to be displayed the user will have to close the component by going back and creating a new instance of the component by reopening the canvas. This is definitely not ideal.

Reply:
tbh, i can solve this by just using useRawData, so is this actually needed? lol

Reply:
@gigadeplex @flo-sch Thank you this is very interesting.

Indeed you can build your own logic using these two functions: `Skia.ImageMakeImageFromEncoded ` and `await Skia.Data.fromURI(uri)`. Our current understanding is that resource loading in React and React Native is a whole science in itself and maybe libraries will do it better than us. All we can provide you are the necessary functions to build these thing.

This is still currently under discussion. stay tuned, in the meantime we would recommend you build our own hooks for this.

Reply:
Okay fair enough, thanks for the answer :)